### ISSUE TITLE
k3s only listen to request from the ipv6 address which is traefik LoadBalancer external-ip

### ISSUE THREAD

**Issue by wanglvhang** [k3s-io/k3s#7541]
**Environmental Info:**
K3s Version: 
k3s version v1.26.4+k3s1 (8d0255af)
go version go1.19.8

Node(s) CPU architecture, OS, and Version: 
Intel(R) Pentium(R) CPU G4400T @ 2.90GHz
22.04.1

Cluster Configuration:
single node cluster with dualstack

**Describe the bug:**
my host has multiple ipv6 global address, when i set my domain to the ipv6 address in external-ip it's ok to visit servcie in k3s, but when i set my domain address to my host address but not external-ip the request will time out.

**Steps To Reproduce:**
deploy k3s on host which has multiple ipv6 address

- Installed K3s: 

**Expected behavior:** **Actual behavior:**
address on host
enp2s0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.31.254  netmask 255.255.255.0  broadcast 192.168.31.255
        inet6 240e:390:c80:82b0:6029:xxxx:9c52:2de6  prefixlen 64  scopeid 0x0<global>
        inet6 240e:390:c80:82b0:5231:xxxx:f303:d497  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::cae7:d7c0:xxxx:b67b  prefixlen 64  scopeid 0x20<link>
        inet6 240e:390:c80:82b0:xxxx:7e8  prefixlen 128  scopeid 0x0<global>

external-ip
kube-system  traefik         LoadBalancer  10.43.83.184   192.168.31.254,240e:390:c80:82b0:xxxx::7e8 

when i request with 240e:390:c80:82b0:xxxx::7e8, OK
when i request with 240e:390:c80:82b0:5231:xxxx:f303:d497, Timeout
both ipv6 address are ok to ping

i'm not sure if this is expected behavior, if it's expected how can i use ipv6 address to access k3s if the host's ip address is dynamic? because even when i try to get ipv6 address inside k3s pod with internet apis, the ipv6 address returned is not the one in external-ip.

my expected behavior is k3s can listen requests to any of those ipv6 address on my host.

**Additional context / logs:**

**Comment by brandond** [comment]
> my host has multiple ipv6 global address, when i set my domain to the ipv6 address in external-ip it's ok to visit servcie in k3s, but when i set my domain address to my host address but not external-ip the request will time out.

Yes, that's correct. This is covered in the documentation: https://docs.k3s.io/networking#how-servicelb-works
>> If the ServiceLB Pod runs on a node that has an external IP configured, the node's external IP is populated into the Service's status.loadBalancer.ingress address list. Otherwise, the node's internal IP is used.
